client/payment-request/index.js

```diff
@@ -653,7 +653,10 @@ jQuery( ( $ ) => {
 		},
 	};
 
-	wcpayPaymentRequest.init();
+	// We don't need to initialize payment request on the checkout page now because it will be initialized by updated_checkout event.
+	if ( ! wcpayPaymentRequestParams.is_checkout_page ) {
+		wcpayPaymentRequest.init();
+	}
 
```

Comment:
```suggestion
if ( wcpayPaymentRequestParams.is_pay_for_order ) {
	wcpayPaymentRequest.init();
}
```

Reply:
@mdmoore The payment request buttons are also present on product pages. Should we also have a check for product pages using existing variables in the localized script, considering that checkout triggers will not fire there?

`
if ( wcpayPaymentRequestParams.is_pay_for_order || wcpayPaymentRequestParams.is_product_page ) {
`

Reply:
@duotive the `! wcpayPaymentRequestParams.is_checkout_page` condition ensures that the payment request button is initialized on product pages.

Reply:
Nice catch! I'm updating that with `(! wcpayPaymentRequestParams.is_checkout_page || wcpayPaymentRequestParams.is_pay_for_order)`